lib/src/models/comment_impl.dart

```diff
@@ -150,10 +149,56 @@ class MoreComments extends RedditBase with RedditBaseInitializedMixin {
       _comments = await reddit.post(apiPath['morechildren'], data);
 
       if (update) {
-        _comments.forEach((c) => c._submission = _submission);
+        _comments.forEach((c) {
+          c._submission = _submission;
+        });
       }
     }
-    return _comments;
+    return _fillCommentsForests(_comments);
+  }
+
+  List<dynamic> _fillCommentsForests(List<dynamic> fullList) {
+    final first = fullList.first;
```

Comment:
This will throw an exception if `fullList` is empty. I'd remove this check and do it below.

Reply:
Agreed, that makes more sense.